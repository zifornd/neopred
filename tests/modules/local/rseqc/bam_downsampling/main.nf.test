nextflow_process {

    name "Test Process RSEQC_BAM_DOWNSAMPLING"
    script "modules/local/rseqc/bam_downsampling/main.nf"
    process "RSEQC_BAM_DOWNSAMPLING"

    test("Should run without failures") {

        when {
	// options '-stub'
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
               input[0] = Channel.of([ [ id:'test' ], file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true), file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)])
               input[1] = Channel.of([ [ id: 'test'], file('$projectDir/tests/test_result/test.paired_end.sorted.stats', checkIfExists: true) ])

                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
