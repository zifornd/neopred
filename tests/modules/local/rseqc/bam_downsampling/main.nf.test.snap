{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test._downsampling.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e0c178468b72f8f864831d86da0b07b4"
                ],
                "downsample_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test._downsampling.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e0c178468b72f8f864831d86da0b07b4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-09T12:57:07.61729"
    }
}